MS Teams Upload attachment feature

vaibhav-db opened the pull request.

vaibhav-db committed:
Chat button is request

vaibhav-db committed:
Added comments in files

pranav-biyani-db committed:
SYMPHONYP-1069-inmemory retry logger added

vaibhav-db committed:
SYMPHONYP-1106 upload attachment feature in famework

vaibhav-db committed:
Pull request #47: Feature/SYMPHONYP-1106 download feature in spring bot team framework

Merge in SYMPHONYP/symphony-java-toolkit from feature/SYMPHONYP-1106-download-feature-in-spring-bot-team-framework to spring-bot-master-db

* commit '15451c97348710774d7276e65acacbba9271d0f1':
  SYMPHONYP-1106 upload attachment feature in famework
  Fix for #342
  update pom
  Update CHANGES.md
  Update CHANGES.md

robmoffat commented on libs/chat-workflow/src/main/java/org/finos/springbot/workflow/java/resolvers/ChatButtonWorkflowResolverFactory.java:
```diff
@@ -0,0 +1,46 @@
+import org.finos.springbot.workflow.java.mapping.ChatHandlerExecutor;
+
+/**
+ * Get chat button detail in controller 
```
Can you describe what this is for?

vaibhav-db replied:
As we discussed last time, we need button information on controller method. For that we have added this class.

robmoffat commented on libs/teams/teams-chat-workflow-spring-boot-starter/src/main/java/org/finos/springbot/teams/messages/FileActivityHandler.java:
```diff
@@ -0,0 +1,148 @@
+import com.microsoft.bot.schema.teams.FileConsentCardResponse;
+import com.microsoft.bot.schema.teams.FileInfoCard;
+
+public class FileActivityHandler extends MessageActivityHandler {
```
Please check the code coverage as a result of adding this.  I have a feeling there will need to be some unit tests added to make sure this works.

vaibhav-db replied:
Sure, I will check the code coverage and also add the unit testcases.

vaibhav-db replied:
committed testcases..

vaibhav-db replied:
Hi @robmoffat ,

Now code coverage is 72.53%

robmoffat replied:
Take a look at this one: https://app.codecov.io/gh/finos/spring-bot/blob/spring-bot-develop/libs%2Fteams%2Fteams-chat-workflow-spring-boot-starter%2Fsrc%2Fmain%2Fjava%2Forg%2Ffinos%2Fspringbot%2Fteams%2Fmessages%2FFileActivityHandler.java

robmoffat reviewed (changes requested):
Looking good so far!

vaibhav-db committed:
SYMPHONYP-1118 test case for File handler

vaibhav-db committed:
Pull request #48: SYMPHONYP-1118 test case for File handler

Merge in SYMPHONYP/symphony-java-toolkit from feature/SYMPHONYP-1106-download-feature-in-spring-bot-team-framework to spring-bot-master-db

* commit '9129c834f6119ed03b861233bb6c6dc1e1ff4b08':
  SYMPHONYP-1118 test case for File handler

vaibhav-db committed:
SYMPHONYP-1118 test case for File handler

vaibhav-db committed:
SYMPHONYP-1118 test case for File handler

vaibhav-db committed:
SYMPHONYP-1118 test case for File handler

vaibhav-db committed:
Pull request #49: Feature/SYMPHONYP-1106 download feature in spring bot team framework

Merge in SYMPHONYP/symphony-java-toolkit from feature/SYMPHONYP-1106-download-feature-in-spring-bot-team-framework to spring-bot-master-db

* commit '4958a388f0b9aa305b75a40a0beddee0275de69c':
  SYMPHONYP-1118 test case for File handler
  SYMPHONYP-1118 test case for File handler
  SYMPHONYP-1118 test case for File handler

vaibhav-db committed:
Release note

vaibhav-db committed:
Release note

vaibhav-db committed:
Pull request #50: Feature/SYMPHONYP-1106 download feature in spring bot team framework

Merge in SYMPHONYP/symphony-java-toolkit from feature/SYMPHONYP-1106-download-feature-in-spring-bot-team-framework to spring-bot-master-db

* commit '224316daf824e9ee8ffb34a1e7583a2e797605e4':
  Release note
  Release note

vaibhav-db committed:
SYMPHONYP-1126 converting spring bot 2.7.10 to spring boot 3.1.8

vaibhav-db committed:
SYMPHONYP-1126 migrate-springbot-framework-from-spring-2.7.1-to-spring-3.1.8
